src/Service/ShopServiceUninstaller.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Shopware\ServiceBundle\Service;
+
+use Psr\Log\LoggerInterface;
+use Shopware\App\SDK\AppConfiguration;
+use Shopware\App\SDK\HttpClient\ClientFactory;
+use Shopware\App\SDK\Shop\ShopRepositoryInterface;
+use Shopware\ServiceBundle\Entity\Shop;
+
+class ShopServiceUninstaller
+{
+    /**
+     * @param ShopRepositoryInterface<Shop> $shopRepository
+     */
+    public function __construct(
+        private readonly ShopRepositoryInterface $shopRepository,
+        private readonly AppConfiguration $serviceConfiguration,
+        private readonly ClientFactory $shopHttpClientFactory,
+        private readonly LoggerInterface $logger,
+    ) {}
+
+    public function run(string $shopId): void
+    {
+        /** @var Shop|null $shop */
+        $shop = $this->shopRepository->getShopFromId($shopId);
+
+        if (null === $shop) {
+            throw new \RuntimeException(sprintf('Shop with id "%s" not found', $shopId));
+        }
+
+        $this->logger->info(sprintf('Uninstalling service for shop "%s"', $shop->getShopId()));
+
+        $client = $this->shopHttpClientFactory->createSimpleClient($shop);
+
+        $result = $client->post($shop->getShopUrl() . '/api/service/uninstall/' . $this->serviceConfiguration->getAppName(), []);
+
+        if ($result->ok()) {
```

Comment:
what happens if the uninstallation fails? should the shop/service be notified otherwise?

Reply:
I'm not entirely sure what we can do, if the response is not 200, we can only try again really. When it doesn't work the shop will still be listed as active on that version so the `services:can-cleanup-version 6.6.2.0` command would fail.